HealthcareSystem/PatientWebApp/Controllers/SurveyController.cs

```diff
@@ -44,6 +48,8 @@ public ActionResult AddSurvey(SurveyDTO surveyDTO)
                 _examinationValidator.CheckIfSurveyAboutExaminationIsCompleted(surveyDTO.ExaminationId);
                 _surveyService.AddSurvey(SurveyMapper.SurveyDTOToSurvey(surveyDTO));
                 _examinationService.CompleteSurveyAboutExamination(surveyDTO.ExaminationId);
+                if(!_examinationService.GetExaminationById(surveyDTO.ExaminationId).PatientCard.PatientJmbg.Equals(HttpContext.User.FindFirst("Jmbg").Value))    
+                    return BadRequest("Patient can only fill out the survey for their own examinations.");
```

Comment:
This should happen before the service is called (but all these methods will be refactored soon so it can be moved around then).